resqpy/olio/uuid.py

```diff
@@ -194,3 +197,16 @@ def version_string(uuid_obj):
     if len(v_str) > max_version_string_length:
         v_str = v_str[:max_version_string_length]
     return v_str
+
+
+def is_uuid(uuid_obj):
+    """Returns boolean indicating whether uuid_obj seems to be a uuid."""
+
+    if isinstance(uuid_obj, uuid.UUID):
+        return True
+    if not uuid_obj or not isinstance(uuid_obj, str):
+        return False
+    if uuid_obj[0] == '_':
+        return len(uuid_obj) == 37
+    return len(uuid_obj) == 36
+    # could also check for hyphens in correct places, and hexadecimals only
```

Comment:
Agreed, I'm sure there must be some code somewhere that can validate a UUID fully.

Reply:
Yes, though I'm happy to let this go for now, as it is unlikely that something like-a-valid-UUID-but-not-quite will show up.